connector/blobuploadconnector/metadata.yaml

```diff
@@ -0,0 +1,21 @@
+type: blobuploadconnector
+scope_name: otelcol/blobuploadconnector
+
+status:
+  class: connector
```

Comment:
connectors are generally only used when we need to convert between different telemetry types (e.g. traces to logs).  Consider making this a processor unless we have use-cases where we need to convert between the signal types.

Reply:
Thanks for the review @dashpole . 

My initial approach was to use a processor, but I ran into issues around data ownership (because I wanted to modify the data). Additionally, since this doesn't do pure computation but also does I/O to upload the data, it wasn't clear that a processor was the right fit. @jsuereth had suggested a connector as an approach to consider.

You can see some of the relevant history in https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/33737 ; in particular, https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/33737#issuecomment-2225861351 from July 12th shows where I was working on it as a processor (in a development branch named [`blob_writer_span_processor`](https://github.com/michaelsafyan/open-telemetry.opentelemetry-collector-contrib/tree/blob_writer_span_processor) ) before switching to the connector approach.

Is it OK with you if we proceed with the connector solution? [It sounds like this feedback was an optional rather than mandatory piece of feedback, but wanted to confirm].

Reply:
Processors can modify data (just make sure to set the "mutates data" capability to true).  If you do not have a cross-signal use case then this needs to be a processor.  This is blocking feedback.

Reply:
Thanks. I've updated the code to use a processor.